GitHub App trigger options new page (#9198)
circleci/circleci-docs: jekyll/_cci2/github-app-trigger-options.adoc

---
contentTags:
  platform:
  - Cloud
---

> = GitHub App trigger options

Reviewer: should this be "trigger event options"?

Reviewer: Cheeky ask here, but somewhere in this or other pages, could we get a header with "run pipelines non non-push events", for SEO?

:page-layout: classic-docs
:page-liquid:
:page-description: A guide to the options available for triggering GitHub App pipelines
:icons: font
:experimental:

> When you create a trigger for a GitHub App pipeline, you can choose from a set of trigger options that correspond to link:https://docs.github.com/en/webhooks/webhook-events-and-payloads[GitHub App events].

Reviewer: ```suggestion
When you create a trigger for a GitHub App pipeline, you can choose from a set of trigger options that are based on link:https://docs.github.com/en/webhooks/webhook-events-and-payloads[GitHub App events] and their properties.
```

== Introduction

> When you set up a project in CircleCI you will also set up your first pipeline and trigger. This first "default" trigger is an "all push" trigger, which corresponds to the link:https://docs.github.com/en/webhooks/webhook-events-and-payloads#push[GitHub App push event]. You can also set up additional triggers for your project during setup, or later in your menu:Project settings[Triggers].

Reviewer: > When you set up a project in CircleCI
I think we should also think about OAuth customers here, and say something like
```suggestion
When you set up your first GitHub App pipeline, either through Project Settings or through the project creation flow, you will also set up your first trigger for your pipeline. Triggers define on what events and conditions your pipeline should run. 
When setting up a GitHub App trigger, you will be presented with a range of trigger event options. The default option is to trigger a pipeline on "all pushes", which corresponds to the link:https://docs.github.com/en/webhooks/webhook-events-and-payloads#push[GitHub App push event], but this can be changed to any of the alternative options in the dropdown. 
```

Reviewer: I don't actually think this suggestion is good, but I wanted to make sure that classic org users don't start reading this and think "hold on it's not me, I'm not creating a new project"

Reviewer: Also - I want to make it clear that during project onboarding they can choose a different option in the dropdown

Author: I'm going to add a screenshot of the menu in the create project flow and change the wording a bit

Once your project is set up you can create additional pipelines and triggers (menu:Project settings[Triggers]). You can configure your triggers to initiate pipelines on a variety of events, including non-push events, for example, a pull request being opened, or refined push events, for example, only pushes to a default branch, or pushes to a tag.

== Scope of push event support

* Pipelines are not triggered on branch or tag deletion push events, even when you have the "All push" trigger option selected.
* Pipelines are not triggered when a new branch is created if there are no new commits on the branch.

> 

Reviewer: I wonder if it's worth transforming this into a "Note". 
I say this because this has actually always been the behaviour of circleci OAuth historically , and afaik noone has ever noticed or complained .

Could be something like:
"Note: regardless of the chosen trigger option, CircleCI never runs pipelines on branch or tag deletion push events, not when a new branch is created if there are no new commits on the branch

== Supported trigger events

> GitHub App triggers allow you to define the work to do (pipeline) when an event takes place (trigger). GitHub App triggers are an orchestration tool available in CircleCI. You can access GitHub App triggers using the "Run on" menu when setting up a trigger. The following preset trigger options are available:

Reviewer: I think we need a name of them. I'd go "trigger rules" because that's how we call them in the API. 
```suggestion
GitHub App triggers allow you to define the work to do (pipeline) when an event takes place (trigger). GitHub App triggers are an orchestration tool available in CircleCI. You can access GitHub App trigger rules using the "Run on" menu when setting up a trigger. The following preset trigger options are available:
```

* *PR opened or pushed to, default branch pushes, tag pushes*: Trigger a pipeline on pushes to a branch for which there is an open Pull Request (PR), when a PR is opened, pushes to any tag, and pushes to the default branch.
* *PR opened*: Trigger a pipeline when a PR is opened only. See the link:https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=opened#pull_request[GitHub docs for the pull request opened event].
* *PR merged*: Trigger a pipeline when a PR is merged only. See the link:https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=closed#pull_request[GitHub docs for pull request closed].
* *PR marked ready for review*: Trigger a pipeline when a PR is taken out of draft mode and marked ready for review. See the link:https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=review_requested#pull_request[GitHub docs for pull request review requested].
* *"run-ci" label added to PR*: Trigger a pipeline when a label "run-ci" is added to a PR. See the link:https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=labeled#pull_request[GitHub docs for pull request labelled].

> 

Reviewer: Can we add a version of this (sorry if you've added elsewhere, havent' read everything yet)

> Different trigger rules cannot be combined in a single trigger. However, you can create multiple triggers for the same pipeline, with each trigger using a different trigger rule.

Author: Added as an FAQ at the end
